### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by diredeaths**
I wanted to create some more structure around the backend. Let me know what you think, what you want to change, etc...

**Some notable changes to directories**

Config: more robust, can be expanded to hold whatever configurations we need.

Loaders: Added a new folder that contains all our logic for loading different types of dependencies as well as express.

Services: Added a gas service

Interfaces: Added this folder so we can organize our custom interfaces

Controllers: I vote to remove this directory. I think it is a bit redundant. We can preform business logic in services that are called from the routes.

**Added a few dependencies**

typedi: Used for dependency injection and organization. It will be really useful when we need to test.

axios: Used for external http/https requests.

body-parser: JSON formatting

winston: better logging

### CODE REVIEW

File: src/loaders/express.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    res.status(404);
+    next(err);
+  });
+
+  /// error handlers
+  app.use((err: any, req: Request, res: Response, next: NextFunction) => {
+    /**
+     * Handle 401 thrown by express-jwt library
+     */
+    if (err.name === 'UnauthorizedError') {
```
Comment (nuurcodes) [inline src/loaders/express.ts:48]: Don't think this is needed as we don't have user authorization

Comment (diredeaths) [reply src/loaders/express.ts:48]: Good call, i'll remove this.

**Review by nuurcodes — APPROVED** [review]
LGTM 💯 
We can remove the controllers in another PR if we feel they are useless

File: src/config/index.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  // This error should crash whole process
+  throw new Error("⚠️  Couldn't find .env file  ⚠️");
+}
+
+export default {
+  /**
+   * Your favorite port
+   */
+  port: parseInt(process.env.PORT || '3000', 10),
+
```
Comment (sal-ash) [inline src/config/index.ts:17]: front end also runs on 3000, so maybe worth changing

**Pull request merged**